### ISSUE TITLE
Enhancement: AI/ML integration/E2E tests on CI

### ISSUE THREAD

**Issue by lukekim** [spiceai/spiceai#3009]
## Goal-State/What/Result

The following AI/LLM functionality is covered by integration and end-to-end (E2E) tests, allowing us to release and update it with high confidence, while preventing and detecting bugs and regressions:
  - AI/LLM HTTP Api
  - Task History and Metrics when using AI/LLM
  - Chat functionality (OpenAI,  Anthropic, supported Local Models / Huggingface)
  - Embeddings and vector search functionality
  - Defining AI/LLM via Spicepod

Out of scope:
 - Covering traditional ML functionality by tests is not a priority (out of scope)
 - No additional reporting/alerting/dashboarding is in scope - test are run on PR and as part of release process.

## Why/Purpose

Ensuring that AI/LLM functionality works correct is absolutely crucial. Currently, this is done mostly manually, which is costly and can lead to undetected bugs. Comprehensive AI/ML testing ensures high-quality AI/LLM functionality, allowing us to detect issues earlier, release with high confidence, and move quickly.

## By When

**Issue/Spec written and reviewed:** 11/05
M1 Integration tests for core LLM functionality - 11/19
M2 E2E tests for core LLM functionality - 11/20
**Done-Done:**  11/20

## Done-Done

- [x] [First Principles Driven](https://github.com/spiceai/spiceai/blob/trunk/docs/PRINCIPLES.md)
- [x] The Algorithm
- [x] PM/Design Review
- [x] DX/UX Review
- [x] Threat Model / Security Review
- [x] Tests
- [x] Performance / Benchmarks
- [x] Documentation
- [x] Samples/Quickstarts/Tutorials
- [x] Release Notes (not required)

## The Algorithm

- [x] Every requirement questioned?
- [x] Delete (Scope) any part you can.
- [x] Simplify.
- [x] Break down into smaller iterations/milestones.
- [x] Opportunities for automation.

### M1 Integration tests include core LLM functionality
 - [x] HTTP Api: search, nsql, chat, embeddings
 - [x] Local models downloading/initialization (huggingface)
 - [x] Model inference (chat)
    - [x] OpenAI
    - [x] HF
    - [x] Anthropic
    - [x] Supported local model types
 - [x] Tools: ListDatasetsTool, SqlTool, TableSchemaTool, DocumentSimilarityTool
 - [x] Embeddings: creation/chunking/search
 - [x] Task History when using AI/LLM
 - [x] Tests are run as part of GH workflow

### M2 E2E tests for core LLM functionality

The following scenarios are covered
 - [x] Spicepod with API based model (OpenAI) to chat w/ dataset with embeddings based on API based model embeddings
 - [x] ~Spicepod with huggingface model to chat w/ dataset with embeddings generated by HF model~ (moved to https://github.com/spiceai/spiceai/issues/3645)
 - [x] Tests use CLI tools: `spice search`, `spice chat`

### M1 Integration tests include core LLM functionality

Update integration tests to build spice with models, add `integration-models` tests to run in parallel with existing integration tests (share the same binaries).

### M2 E2E tests for core LLM functionality

Use similar approach as for [E2E Test CI](https://github.com/spiceai/spiceai/actions/workflows/e2e_test_ci.yml)

**Comment by sgrebnov** [comment]
11/10 update: 
 - On track to complete tests by 11/18
 - Target outcome for 11/18 release: done-done

**Comment by lukekim** [comment]
@sgrebnov please add missing sub-tasks and add to the weekly board.

**Comment by sgrebnov** [comment]
11/17 update:
1. Made progress but this is still in progress (was not able to fully focus on this task)
2. Target completeness date has been **moved on Wed 11/20**
3. Target outcome by the end of the week? Done-done

**Comment by sgrebnov** [comment]
Status (11/25)
 - To be finished 11/25 (Mon). Remaining work - add tests for local models
 - Created separate work item to [Run HF and local model tests on a self-hosted GitHub runner ](https://github.com/spiceai/spiceai/issues/3645) - tests exist and can be run locally but require special runner

**Comment by peasee** [comment]
Done-Done: today

**Comment by sgrebnov** [comment]
The enhancement is done
